Firebase.agent.lib.nut

```diff
@@ -569,43 +613,50 @@ class Firebase {
         if (_debug) server.error(message);
     }
 
-    // return a Promise if the Promise library is included
-    function _returnPromise (request){
-        if (_promiseIncluded) {
-            return Promise(function (resolve,reject){
-                    request.sendasync(function(res){
-                        local data = null ;
-                        try {
-                            data = http.jsondecode(res.body);
-                            if ( 200 <= res.statuscode && res.statuscode < 300) {
-                                resolve(data);
-                            } else {
-                                reject(data.error);
-                            }
-                        } catch (err){
-                            reject (err);
-                        }
-                    }.bindenv(this));
-            });
-        }
-        return;
+    // return a Promise 
+    function _createRequestPromise(request) {
+        return Promise(function (resolve, reject) {
+            request.sendasync(_createResponseHandler(resolve, reject));
+        });
     }
 
     // process the http response accordingly
-    function _processResponse (request,callback) {
-        request.sendasync(function(res) {
-            local data = res.body ;
+    function _sendRequest(request, callback) {
+        local resolve = function (data) { 
+            callback && callback(null, data);
+        }; 
+        local reject = function (err) { 
+            callback && callback(err, null);
+        }; 
+        request.sendasync(_createResponseHandler(resolve, reject));
+    }
+
+    function _createResponseHandler(resolve, reject) { 
```

Comment:
Just for this specific method I'd rename the arguments to `onSuccess`, `onError` respectively. As it has nothing to do with the Promises.